### PR TITLE
[RISCV] Simplify (srl (and X, Mask), Const) to TH_EXTU

### PR DESCRIPTION
**Pull request by MouseSplinter**

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp [hunk 1]
```diff
@@ -1164,6 +1164,15 @@ void RISCVDAGToDAGISel::Select(SDNode *Node) {
     }
 
     unsigned LShAmt = Subtarget->getXLen() - TrailingOnes;
+    if (Subtarget->hasVendorXTHeadBb() && TrailingOnes > ShAmt) {
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp:1167]: TrailingOnes > ShAmt is always true here, it was checked on line 1142.

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp (hunk 1 above)
Comment (MouseSplinter) [reply llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp]: Thanks! done

**Comment by topperc** [comment]
@MouseSplinter do you have commit access?

**Comment by MouseSplinter** [comment]
> @MouseSplinter do you have commit access?

Yes, but I notice a strange [CI fail](https://buildkite.com/llvm-project/github-pull-requests/builds/90710#019145c2-81cd-4dc4-8500-e443d5746345) , do you know how to resolve it?

**Comment by topperc** [comment]
> > @MouseSplinter do you have commit access?
> 
> Yes, but I notice a strange [CI fail](https://buildkite.com/llvm-project/github-pull-requests/builds/90710#019145c2-81cd-4dc4-8500-e443d5746345) , do you know how to resolve it?

No. Maybe just merge from `main` and restart the CI.

**Pull request merged**